HighwayPatrol.cpp

```diff
@@ -7,11 +7,13 @@ HighwayPatrol::HighwayPatrol() : Vehicle("HighwayPatrol")
 
 }
 
+HighwayPatrol::~HighwayPatrol() = default;
+
 void HighwayPatrol::scanHighway(Highway* h)
 {
     std::cout << name << ": scanning highway for speeders" << std::endl;
 
-    for( size_t i = 0; i < h->vehicles.size(); ++i  )
+    for( size_t i = h->vehicles.size(); --i;  )
```

Comment:
You're not comparing `i` with anything to make it stop looping!!   See `#general`, I recently shared a link showing the different results of pre/post decrementing
- [ ] revise.

Reply:
Hmm.  --i >= 0 kept giving me a warning and crashing.  --i > 0  better? This will give us 543210 (which is all 6 vehicles).  That right?